src/Data/YAML/Event/Writer.hs

```diff
@@ -124,58 +123,115 @@ putNode = \docMarker -> go (-1 :: Int) (not docMarker) BlockIn
+            | FlowOut  <- c = FlowIn
+            | BlockKey <- c = FlowKey
+            | FlowKey  <- c = FlowKey
+            | otherwise = error "Invalid context Mapping Flow style"
+
+          n' = n + 1
+
+          doEol = case c of
+            FlowKey -> mempty
+            FlowIn -> mempty
+            _        -> eol
+
+          g (MappingEnd : rest) = eol <> (if sol then mempty else ws) <> mkInd (n - 1) <> "}"<> doEol <> cont rest
+          g ys                  = "," <> eol<> mkInd n'<> putKey ys (\zs -> go n' False FlowIn zs g)
+
+          putKey zs cont2
+            | isSmallKey zs =    go n' (n == 0) FlowKey zs (\ys -> ":" <> cont2 ys)
+            | otherwise     = "?" <> go n' False FlowIn zs (\ys ->eol<> mkInd n' <> ":" <> cont2 ys)
+
+
+    goSeq _ sol _ anc tag _ (SequenceEnd : rest) cont = pfx $ "[]\n" <> cont rest
       where
-        g (MappingEnd : rest) = cont rest
-        g ys                  = pfx <> putKey ys (\zs -> go n False BlockOut zs g)
+        pfx cont' = (if sol then mempty else ws) <> anchorTag'' (Right ws) anc tag cont'
 
-        pfx = mkInd n
+    goSeq n sol c anc tag sty xs cont = case sty of 
+      Block -> case c of      
+          BlockOut -> anchorTag'' (Left ws) anc tag (eol <> mkInd n' <> "-" <> go n' False c' xs g)
 
-        putKey zs cont2
-          | isSmallKey zs =    go n (n == 0) BlockKey zs (\ys -> ":" <> cont2 ys)
-          | otherwise     = "?" <> go n False BlockIn zs (\ys -> mkInd n <> ":" <> cont2 ys)
+          BlockIn
+            | not sol && n == 0 {- "---" case -} -> goSeq n sol BlockOut anc tag sty xs cont
+            | otherwise -> (if sol then mempty else ws) <> anchorTag'' (Right (eol <> mkInd n')) anc tag ("-" <> go n' False c' xs g)
 
+          BlockKey -> error ("sequence in block-key context not supported")
 
+          _ -> error "Invalid Context in Block style"
 
-    goSeq n sol c anc tag sty (SequenceEnd : rest) cont = pfx $ "[]\n" <> cont rest
-      where
-        pfx cont' = (if sol then mempty else ws) <> anchorTag'' (Right ws) anc tag cont'
+        where
+          c' = BlockIn
+          n' | BlockOut <- c = max 0 (n - 1)
+             | otherwise     = n
 
-    goSeq n sol c anc tag sty xs cont = case c of
-        BlockOut -> anchorTag'' (Left ws) anc tag (eol <> mkInd n' <> "-" <> go n' False c' xs g)
+          g (SequenceEnd : rest) = cont rest
+          g ys                   = mkInd n' <> "-" <> go n' False c' ys g
 
-        BlockIn
-          | not sol && n == 0 {- "---" case -} -> goSeq n sol BlockOut anc tag sty xs cont
-          | otherwise -> (if sol then mempty else ws) <> anchorTag'' (Right (eol <> mkInd n')) anc tag ("-" <> go n' False c' xs g)
+      Flow -> (if sol then mempty else ws) <> anchorTag'' (Right ws) anc tag ("[" <> eol <> mkInd n' <> go n' False c' xs g)
 
-        BlockKey -> error ("sequence in block-key context not supported")
+        where
+            c'| FlowIn   <- c = FlowIn
```

Comment:
see comment about `c'` in `goMap` -- however, at this point I'd say factor out this `case` into a common function definition; this is basically the implementation of

```
[136]	in-flow(c)	::=	c = flow-out  ⇒ flow-in
                                c = flow-in   ⇒ flow-in
                                c = block-key ⇒ flow-key
                                c = flow-key  ⇒ flow-key
```

so I'd call the function `in_flow` or similar; and mention in the docstring that it originates from https://yaml.org/spec/1.2/spec.html#in-flow(c)